src/rules/mock-rule-builder.ts

```diff
@@ -154,6 +155,32 @@ export default class MockRuleBuilder {
         return this.addRule(rule);
     }
 
+    /**
+     * Reply to matched with the given status and a JSON object, with optionally more headers.
+     *
+     * This method is shorthand for:
+     * server.get(...).thenReply(status, JSON.stringify(object), { 'Content-Type': 'application/json' })
+     *
+     * This method returns a promise that resolves with a mocked endpoint.
+     * Wait for the promise to confirm that the rule has taken effect
+     * before sending requests to be matched. The mocked endpoint
+     * can be used to assert on the requests matched by this rule.
+     */
+    thenJSON(status: number, data?: object, headers?: OutgoingHttpHeaders): Promise<MockedEndpoint> {
+        if (headers == null) {
+            headers = {};
+        }
+        merge(headers, { 'Content-Type': 'application/json' });
```

Comment:
This should merge into a new object, not mutate the headers object you pass in, since that'll be super weird if it's not a literal.